src/net_processing.cpp

```diff
@@ -1994,7 +1994,7 @@ void PeerManagerImpl::BlockConnected(
 
     // The following task can be skipped since we don't maintain a mempool for
     // the ibd/background chainstate.
-    if (role == ChainstateRole::BACKGROUND) {
+    if (is_ibd || role == ChainstateRole::BACKGROUND) {
```

Comment:
is the comment (already mentioning IBD) still accurate?

Reply:
> is the comment (already mentioning IBD) still accurate?

The comment was accurate, but the code didn’t behave as described. Now it does.